openelevationservice/server/api/querybuilder.py

```diff
@@ -32,19 +32,95 @@ def _getModel(dataset):
+        
+    :returns: 3D polygon as GeoJSON or WKT
+    :rtype: string
+    """
+    Model = _getModel(dataset)
+
+    # May be a good idea to make this a parameter from request
+    num_ranges = 23
+    range_div = 43
+    
+    if geometry.geom_type == 'Polygon':
+        query_geom = db.session \
+                            .query(func.ST_SetSRID(func.ST_PolygonFromText(geometry.wkt), 4326) \
+                            .label('geom')) \
+                            .subquery().alias('pGeom')
+
+        result_pixels = db.session \
+                            .query(func.DISTINCT(func.ST_PixelAsPolygons(
+                                func.ST_Clip(Model.rast, query_geom.c.geom, 0), #.geom
+                                1, False))) \
+                            .select_from(query_geom.join(Model, ST_Intersects(Model.rast, query_geom.c.geom))) \
+                            .all()
+        
+        polygon_col, height_col = format_PixelAsGeoms(result_pixels)
+
+        column_set = db.session \
+                            .query(polygon_col.label("geometry"),
+                                   func.LEAST(func.floor(height_col / range_div), num_ranges).label("colorRange")) \
+                            .subquery().alias('columnSet')
+
+        query_features = db.session \
+                            .query(func.jsonb_build_object(
+                                'type', 'Feature',
+                                'geometry', func.ST_AsGeoJson(
+                                    func.ST_SimplifyPreserveTopology(func.ST_Union(func.array_agg(
+                                        func.ST_ReducePrecision(column_set.c.geometry, 1e-12)
+                                    )), 1e-12)
+                                ).cast(JSON),
+                                'properties', func.json_build_object(
+                                    'heightBase', column_set.c.colorRange * range_div,
+                                )).label('features') \
+                            ).select_from(column_set) \
+                            .group_by(column_set.c.colorRange) \
+                            .subquery().alias('rfeatures')
+
+        # Return GeoJSON directly in PostGIS
+        query_final = db.session \
+                            .query(func.jsonb_build_object(
+                                'type', 'FeatureCollection',
+                                'features', func.jsonb_agg(query_features.c.features))) \
+                            .select_from(query_features)
+
+    else:
+        raise InvalidUsage(400, 4002, "Needs to be a Polygon, not a {}!".format(geometry.geom_type))
+    
+    result_geom = query_final.scalar()
+
+    # Behaviour when all vertices are out of bounds
+    if result_geom == None:
+        raise InvalidUsage(404, 4002,
+                           'The requested geometry is outside the bounds of {}'.format(dataset))
+    
+    return result_geom
+
+
```

Comment:
If 'polygon_elevation(geometry, format_out, dataset)' is no longer to be used for anything else, then replace its implementation by that of 'polygon_coloring_elevation'.